client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js

```diff
@@ -1,170 +1,205 @@
+            })))
+        .then(() => {
+          this.TucToast.success(this.$translate.instant('telephony_alias_configuration_agents_delete_success'));
+          return this.fetchAgentsIds().then((ids) => {
+            this.agents.ids = ids;
+          });
+        })
+        .catch((err) => {
+          this.TucToast.error([this.$translate.instant('telephony_alias_configuration_agents_delete_error'), _.get(err, 'data.message')].join(' '));
+          return this.$q.reject(err);
+        })
+        .finally(() => {
+          this.agents.isDeleting = false;
+          this.agents.selected = {};
+        });
+    }
+
+    static startEdition(agent) {
+      _.set(agent, 'inEdition', _.pick(agent, ['description', 'number', 'simultaneousLines', 'status', 'timeout', 'type', 'wrapUpTime']));
+    }
+
+    static isValidAgent({
+      number, timeout, simultaneousLines, wrapUpTime,
+    }) {
+      let valid = true;
+      valid = valid && number;
+      valid = valid && /^\d{1,6}$/.test(timeout);
+      valid = valid && /^\d{1,6}$/.test(wrapUpTime);
+      valid = valid && /^([1-9]|10)$/.test(simultaneousLines);
+      return valid;
+    }
+
+    updateAgent(agent) {
+      _.set(agent, 'isUpdating', true);
+      return this.OvhApiTelephony.OvhPabx().Hunting().Agent().v6()
+        .change({
+          billingAccount: this.$stateParams.billingAccount,
+          serviceName: this.$stateParams.serviceName,
+          agentId: agent.agentId,
+        }, agent.inEdition).$promise
+        .then(() => {
+          _.assign(agent, agent.inEdition);
+          _.set(agent, 'inEdition', null);
+          this.TucToast.success(this.$translate.instant('telephony_alias_configuration_agents_update_success'));
+        })
+        .catch((err) => {
+          this.TucToast.error([this.$translate.instant('telephony_alias_configuration_agents_update_error'), _.get(err, 'data.message')].join(' '));
+          return this.$q.reject(err);
+        })
+        .finally(() => {
+          _.set(agent, 'isUpdating', false);
+        });
+    }
+
+    cancelAddAgent(pos) {
+      if (pos === 0 && this.addAgentForm.numbers.length > 1) {
+        this.addAgentForm.numbers.shift();
+      } else if (this.addAgentForm.numbers.length > 1) {
+        _.pullAt(this.addAgentForm.numbers, pos);
+      } else {
+        this.addAgentForm.numbers[0] = null;
+      }
+    }
+
+    // Check if external numbers are defined into the list
+    checkExternalNumber() {
+      const filter = this.addAgentForm.numbers.filter(number => this.groupList
+        .filter(telgroup => telgroup.getAllServices()
+          .filter(service => service.serviceType === 'alias')
+          .filter(service => service.serviceName === number)
+          .length > 0)
+        .length > 0);
+      return filter.length !== this.addAgentForm.numbers.length;
+    }
+
+    addAgents() {
+      if (this.checkExternalNumber()) {
+        const modal = this.$uibModal.open({
```

Comment:
You can get rid of `modal` var : 
```suggestion
        this.$uibModal.open(myModal).result.then(() => etc....)
```